data-prepper-plugins/parse-json-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/parse/json/ParseJsonProcessorTest.java

```diff
@@ -125,6 +130,44 @@ void test_when_differentSourceAndDestination_then_processorParsesCorrectly() {
         verifyNoInteractions(handleFailedEventsOption);
     }
 
+    @ParameterizedTest
+    @ValueSource(ints = {0,1,2,3,4,5,6,7,8,9,10})
+    void test_depth_option_with_same_source_and_destination(int depth) throws Exception {
+        final String source = "root_source";
+        when(processorConfig.getSource()).thenReturn(source);
+        when(processorConfig.getDestination()).thenReturn(source);
+        when(processorConfig.getDepth()).thenReturn(depth);
+        parseJsonProcessor = createObjectUnderTest(); // need to recreate so that new config options are used
+
+        Map<String, Object> mapValue = new HashMap<>();
+        Map<String, Object> prevMap = null;
+        Object expectedResult = null;
+        // Create a map with 12 nested levels
+        for (int i = 11; i >= 0; i--) {
```

Comment:
I'm wondering about the complexity of this test. Do you think it would make sense to have some simpler versions to verify depth of 1 or 2?

Reply:
OK. Will add two small tests for testing depth 1 and 2.